### PR TITLE
Add TESTING.md and Makefile

### PR DESCRIPTION
**Pull request by nshalman**
Per feedback in #67 this is the beginning of a TESTING.md and the Makefile that was used to test the changes in #67.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+help:
+	@echo 'Did you mean "make output"?'
+
+docker-image:
+	sudo docker build -t $(BUILDER) .
+	touch docker-image
+
+output: docker-image
+	mkdir $@
+	sudo docker run --name $(CONTAINER) --privileged=true -v $(PWD)/$@:/output $(BUILDER)
```
Comment (byxorna) [inline Makefile]: assuming things about the environment is not cool. how do you know this is not being run on a mac or linux box where the user has write permissions to the docker socket? or that sudo is even the privilege escalation tool in use? please remove the use of sudo and let the underlying tools either succeed or fail due to lack of permissions.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	sudo chown -R $(USER) $@
+
+clean:
+	sudo losetup -a | cut -d: -f1 | xargs -r -n 1 sudo losetup -d
```
Comment (byxorna) [inline Makefile]: losetup doesnt exist on OSX, so this is a pretty huge assumption. Personally I build images in docker for mac and dont need this hack, or need to run it in the hyve VM instead.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	sudo rm -rf output docker-image
+	sudo docker rm $(CONTAINER) || true
+	sudo docker rmi $(BUILDER) || true
+
+run-serial: output
```
Comment (byxorna) [inline Makefile]: I think the test targets should all be prefixed by `test-` so its clear what their intent is (clearer than reading comments in the makefile)

**Review by byxorna — CHANGES_REQUESTED** [review]
Generally looks good, but makes assumptions about the system the makefile is running on. Some tweaks and it should be good to go! Thanks for this changeset!

**Comment by nshalman** [comment]
Not quite where things should be for a Mac (e.g. if on a Mac and Docker VM is not running it will think it needs sudo...)

**Comment by nshalman** [comment]
Well, it's not pretty, but I didn't want to have to resort to some sort of configure script.
Based on my own testing on both a Mac and a Linux box, this should be both defensive about making sure Docker is running, and capable of using sudo only when needed.

The test-* targets are also hidden when not on Linux.

Let me know what you think.

**Comment by roymarantz** [comment]
Generally speaking I'm not wild about Makefile running sudo.  Why not assume that the invoking user has permissions to run docker directly?

**Comment by nshalman** [comment]
Because mine doesn't on my Linux box.

**Pull request closed without merging**